src/Serialization/main.jl

```diff
@@ -53,14 +53,21 @@ end
 # registerSerializationType is a macro to ensure that the string we generate
 # matches exactly the expression passed as first argument, and does not change
 # in unexpected ways when import/export statements are adjusted.
-macro registerSerializationType(ex::Any, str::Union{String,Nothing} = nothing)
+# It also sets the value of serialize_with_id, which determines
+# whether or not the type can be back referenced.
+macro registerSerializationType(
+    ex::Any,
+    uses_id::Bool = false,
+    str::Union{String,Nothing} = nothing)
```

Comment:
Instead of forcing calls that pass a value for `str` to also pass one for `uses_id`, even if it is false, I think you can use "macro overloading: I.e. have two macros with the same name but different argument lists:
```julia-repl
julia> macro foo1(ex::Any, uses_id::Bool = false, str::Union{String,Nothing} = nothing)
         return :(:bar, $ex, $uses_id, $str)
       end
@foo1 (macro with 3 methods)

julia> @foo1 1+2
(:bar, 3, false, nothing)

julia> macro foo1(ex::Any, str::Union{String,Nothing})
         return :(:quux,$ex,false,$str)
       end
@foo1 (macro with 4 methods)

julia> @foo1 1+2
(:bar, 3, false, nothing)

julia> @foo1 1+2 false
(:bar, 3, false, nothing)

julia> @foo1 1+2 "house"
(:quux, 3, false, "house")
```

Reply:
Oh, and those two macro implementations can share code: just move the bulk of the macro implementation into a regular function, then invoke it from both macro variants